### PR TITLE
Changepath

### PR DESCRIPTION
**Pull request by shiminjia**
1. To change import path to "github.com/shiminjia/bookcommunity/"
2. To add .gitignore, .gcloudignore, app.yaml, cloudbuild.yaml and README.md
    app.yaml is the configure for GAE.
    cloudbuild.yaml is the configure for Cloudbuild. 
    When main branch is merged, it will be build and deployed on 
    https://bookcommunity.appspot.com
3. To add middlewares and declaim the router "/" for authorizing user.
    Pleasae read middleware/auth.go to get more information.
4. Some other small modification.

### CODE REVIEW

**Comment by ulip128** [comment]
I prefer fine granularity. It is common.

File: controller/user.go [hunk 1]
```diff
@@ -1,35 +1,77 @@
...
+	if err := c.ShouldBindJSON(&u); err != nil {
 		middleware.ErrorResponse(c, http.StatusBadRequest, config.ErrBind)
 		return
 	}
 
-	data := &model.Token{Token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1MjgwMTY5MjIsImlkIjowLCJuYmYiOjE1M" +
-		"jgwMTY5MjIsInVzZXJuYW1lIjoiYWRtaW4ifQ.LjxrK9DuAwAzUD8-9v43NzWBN7HXsSLfebw92DKd1JQ"}
+	//todo get user info from db by user name
+	//this is a demo now
+	var d = model.UserInfoWithId{
```
Comment (ulip128) [inline controller/user.go]: Demo?

File: controller/user.go [hunk 2]
```diff
@@ -1,35 +1,77 @@
...
+		Id:       "1234",
+		Email:    "zhangsan@gmail.com",
+		Username: "zhangsan",
+		Password: "11111111",
+	}
+
+	//compare the password with request and db
+	if u.Password != d.Password {
```
Comment (ulip128) [inline controller/user.go]: Password must be hashed.

File: main.go [hunk 3]
```diff
@@ -2,30 +2,66 @@ package main
...
+		f, _ := os.Create("gin.log")
+		gin.DefaultWriter = io.MultiWriter(f)
+	} else if mode == "prod" {
+		gin.DisableConsoleColor()
+		//todo need to change the name of log like yyyyMMdd.log
+		f, _ := os.Create("gin.log")
+		gin.DefaultWriter = io.MultiWriter(f)
+	} else {
+		router.Use(gin.Logger())
+	}
```
Comment (ulip128) [inline main.go]: I prefer a switch clause.

File: utils/token.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+	//sign the token
+	tokenString, err = token.SignedString([]byte(secret))
+	return
+}
+
+func VerifyToken(token string) bool {
+
+	return true
+
```
Comment (ulip128) [inline utils/token.go]: remove empty lines.

File: controller/user.go (hunk 1 above)
Comment (shiminjia) [reply controller/user.go]: Yes

File: controller/user.go (hunk 2 above)
Comment (shiminjia) [reply controller/user.go]: will improve in the future.

File: main.go (hunk 3 above)
Comment (shiminjia) [reply main.go]: changed at next version

File: utils/token.go (hunk 4 above)
Comment (shiminjia) [reply utils/token.go]: I will add complete this function at next version.

**Pull request merged**